lib/ts/recipe/session/cookieAndHeaders.ts

```diff
@@ -111,7 +72,71 @@ export function setFrontTokenInHeaders(res: BaseResponse, userId: string, atExpi
 }
 
 export function getCORSAllowedHeaders(): string[] {
-    return [antiCsrfHeaderKey, ridHeaderKey];
+    return [antiCsrfHeaderKey, HEADER_RID, authorizationHeaderKey, authModeHeaderKey];
+}
+
+function getCookieNameFromTokenType(tokenType: TokenType) {
+    switch (tokenType) {
+        case "access":
+            return accessTokenCookieKey;
+        case "refresh":
+            return refreshTokenCookieKey;
+        default:
+            throw new Error("Unknown token type, should never happen.");
+    }
+}
+
+function getResponseHeaderNameForTokenType(tokenType: TokenType) {
+    switch (tokenType) {
+        case "access":
+            return accessTokenHeaderKey;
+        case "refresh":
+            return refreshTokenHeaderKey;
+        default:
+            throw new Error("Unknown token type, should never happen.");
+    }
+}
+
+export function getToken(req: BaseRequest, tokenType: TokenType, transferMethod: TokenTransferMethod) {
+    if (transferMethod === "cookie") {
+        return req.getCookieValue(getCookieNameFromTokenType(tokenType));
+    } else if (transferMethod === "header") {
+        const value = req.getHeaderValue(authorizationHeaderKey);
+        if (value === undefined || !value.startsWith("Bearer ")) {
+            return undefined;
+        }
+
+        return value.replace(/^Bearer /, "");
```

Comment:
what if the user adds "Bearer  <token>" -> that is two spaces between "Bearer" and the token. We should also do trim to the result.